### DISCUSSION TITLE
How to effectively block Bittorrent?

### DISCUSSION THREAD

**Discussion by Heclalava** [v2fly/v2ray-core#1044] [Q&A]
I have tried IPTables rules, host file destination blocking of torrent websites and trackers, I have also tried blocking bittorrent in config.json in outbound with sniffing enabled. Nothing seems to work, I am still able to download torrents. I am getting a lot of abuse complaints recently from VPS providers and need to really shut down torrents. Does anyone actually have an effective method that works?

**Comment by houmie** [comment]
Hi @Heclalava 

I'm in a similar situation as you. Have you found a solution to this problem yet?

I saw your [project](https://github.com/Heclalava/blockpublictorrent-iptables). Beside the fact that it hangs when I run it on Debian 11, I don't think this solution will work, because all it does is preventing the user to download the torrent file.  But users can disable the VPN, download the magnet torrent file, then enable VPN and start downloading. Torrent downloads can also happen Peer-to-peer without the need for a tracker.

This is really a big problem.

**Comment by Heclalava** [comment]
I've changed the nature of my business. Clients get their own private server, unless they have one that allows Torrents, if they do torrent download on their server and the provider doesn't allow it, and I get an abuse complaint I now know exactly who to talk to. I no longer do shared servers anymore.

I still use that script, and IP tables rules, but since I've changed my business model, I'm not getting abuse complaints anymore. Clients know they can get caught now.
